frontend/pages/hosts/details/HostDetailsPage/modals/UnlockModal/UnlockModal.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useContext } from "react";
+import { AxiosError } from "axios";
+import { useQuery } from "react-query";
+
+import { NotificationContext } from "context/notification";
+import hostAPI, { IUnlockHostResponse } from "services/entities/hosts";
+
+import Modal from "components/Modal";
+import Button from "components/buttons/Button";
+import Spinner from "components/Spinner";
+import DataError from "components/DataError";
+
+const baseClass = "unlock-modal";
+
+interface IUnlockModalProps {
+  id: number;
+  platform: string;
+  hostName: string;
+  pin?: number;
+  onSuccess: () => void;
+  onClose: () => void;
+}
+
+const UnlockModal = ({
+  id,
+  platform,
+  hostName,
+  pin,
+  onSuccess,
+  onClose,
+}: IUnlockModalProps) => {
+  const { renderFlash } = useContext(NotificationContext);
+  const [isUnlocking, setIsUnlocking] = React.useState(false);
+
+  const {
+    data: macUnlockData,
+    isLoading: macIsLoading,
+    isError: macIsError,
+  } = useQuery<IUnlockHostResponse, AxiosError>(
+    ["mac-unlock-pin", id],
+    () => hostAPI.unlockHost(id),
+    {
+      enabled: platform === "darwin",
+    }
+  );
+
+  const onUnlock = async () => {
+    setIsUnlocking(true);
+    try {
+      await hostAPI.unlockHost(id);
+      onSuccess();
+      renderFlash("success", "Host Unlocked successfully!");
+    } catch (error) {
+      const err = error as AxiosError;
+      renderFlash("error", err.message);
```

Comment:
Same, are these error states a part of this ticket?
<img width="746" alt="Screenshot 2024-02-12 at 10 49 24 AM" src="https://github.com/fleetdm/fleet/assets/71795832/438f4b1f-4a10-4c9d-9faf-999343d346ed">

Reply:
same. these will come from the API. that's why they are blank atm. I will check that they work correctly when the API is ready